flocker/__init__.py

```diff
@@ -11,27 +11,6 @@
 REST_API_PORT = 4523
 
 
-def _disable_pyrsistent_c_extensions():
-    """
-    Pyrsistent sometimes segfaults. Disabling the C extension reduces the
-    likelihood of this happening.
-
-    We do this first so it happens before pyrsistent is imported.
-
-    In theory this bug is fixed in pyrsistent 0.11.7, so this may no
-    longer be necessary and is likely worth risking for higher
-    performance. Once we have benchmarking framework we can assess
-    risk/benefit ratio better.
-
-    The mechanism for disabling extensions is documented at
-    https://github.com/tobgu/pyrsistent/blob/master/CHANGES.txt#L17-L19.
-    """
-    import os
-    os.environ[b"PYRSISTENT_NO_C_EXTENSION"] = b"1"
-_disable_pyrsistent_c_extensions()
-del _disable_pyrsistent_c_extensions
```

Comment:
I've had flocker running on 58 nodes over the weekend with the C extension enabled and the cluster is still functioning. Just presenting data.

It definitely gave a significant speedup in some situations, and I actually think that unless the segfault is extremely frequent flocker is designed to be impervious to segfaults (assuming systemd would just respawn the daemon). It's only scary if the segfaults are indicative of worse bugs. IMHO, not as bad as the multithreading issues in the drivers.